src/Foundation/NSDate.cs

```diff
@@ -54,7 +63,20 @@ public partial class NSDate {
 			if (dt.Kind == DateTimeKind.Unspecified)
 				throw new ArgumentException ("DateTimeKind.Unspecified cannot be safely converted");
 
-			return FromTimeIntervalSinceReferenceDate ((dt.ToUniversalTime ().Ticks - NSDATE_TICKS) / (double) TimeSpan.TicksPerSecond);
+			var dtUnv = dt.ToUniversalTime ();
+
+			var calendar = new NSCalendar (NSCalendarType.Gregorian);
+			calendar.TimeZone = NSTimeZone.FromName ("UTC");
+			var components = new NSDateComponents {
+				Day = dtUnv.Day, Month = dtUnv.Month, Year = dtUnv.Year, Hour = dtUnv.Hour,
+					Minute = dtUnv.Minute, Second = dtUnv.Second, Nanosecond = dtUnv.Millisecond * 1000000
+			};
+
+			var retDate = calendar.DateFromComponents (components);
+			if (retDate == null)
+				throw new ArgumentOutOfRangeException ($"Value is outside the range of NSDate {dt}");
+
+			return retDate;
 		}
 #else
 		public static implicit operator DateTime (NSDate d)
```

Comment:
Should we update both of these implicit operators as well? Would require probably getting rid of the constant at the top of the file as well.

Reply:
No, the `!XAMCORE_2_0` part can be deleted - we don't build that code anymore.